src/plugins/chart_expressions/expression_heatmap/public/compoments/heatmap_component.tsx

```diff
@@ -113,7 +112,20 @@ function computeColorRanges(
   return { colors, ranges };
 }
 
-export const HeatmapComponent: FC<HeatmapRenderProps> = ({
+const getAccessor = (value: string | ExpressionValueVisDimension, columns: DatatableColumn[]) => {
+  if (typeof value === 'string') {
+    return value;
+  } else {
+    const accessor = value.accessor;
+    if (typeof accessor === 'number') {
+      return columns[accessor].id;
+    } else {
+      return accessor.id;
+    }
+  }
+};
```

Comment:
nitpick: this can be flattened as there's a `return` statement in each branch
```suggestion
  if (typeof value === 'string') {
    return value;
  }
  const accessor = value.accessor;
  if (typeof accessor === 'number') {
    return columns[accessor].id;
  }
  return accessor.id;
};
```